openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -249,38 +267,42 @@ def tearDown(self):
-        [True, COURSE_VISIBILITY_PUBLIC, CourseUserType.GLOBAL_STAFF, True, True],
+        [False, False, COURSE_VISIBILITY_PRIVATE, CourseUserType.ANONYMOUS, True, False],
+        [False, False, COURSE_VISIBILITY_PUBLIC_OUTLINE, CourseUserType.ANONYMOUS, True, False],
+        [False, False, COURSE_VISIBILITY_PUBLIC, CourseUserType.ANONYMOUS, True, False],
+        [False, True, COURSE_VISIBILITY_PRIVATE, CourseUserType.ANONYMOUS, True, False],
+        [True, True, COURSE_VISIBILITY_PUBLIC, CourseUserType.ANONYMOUS, False, True],
+        [True, True, COURSE_VISIBILITY_PUBLIC_OUTLINE, CourseUserType.ANONYMOUS, True, True],
+        [False, True, COURSE_VISIBILITY_PUBLIC, CourseUserType.ANONYMOUS, False, True],
+        [False, True, COURSE_VISIBILITY_PUBLIC_OUTLINE, CourseUserType.ANONYMOUS, True, True],
+
+        [False, False, COURSE_VISIBILITY_PRIVATE, CourseUserType.UNENROLLED, True, False],
+        [False, False, COURSE_VISIBILITY_PUBLIC_OUTLINE, CourseUserType.UNENROLLED, True, False],
+        [False, False, COURSE_VISIBILITY_PUBLIC, CourseUserType.UNENROLLED, True, False],
+        [False, True, COURSE_VISIBILITY_PRIVATE, CourseUserType.UNENROLLED, True, False],
+        [True, True, COURSE_VISIBILITY_PUBLIC, CourseUserType.UNENROLLED, False, True],
+        [True, True, COURSE_VISIBILITY_PUBLIC_OUTLINE, CourseUserType.UNENROLLED, False, True],
+        [False, True, COURSE_VISIBILITY_PUBLIC, CourseUserType.UNENROLLED, True, True],
+        [False, True, COURSE_VISIBILITY_PUBLIC_OUTLINE, CourseUserType.UNENROLLED, True, True],
+
+        [False, False, COURSE_VISIBILITY_PRIVATE, CourseUserType.ENROLLED, False, True],
+        [False, True, COURSE_VISIBILITY_PRIVATE, CourseUserType.ENROLLED, False, True],
+        [False, True, COURSE_VISIBILITY_PUBLIC_OUTLINE, CourseUserType.ENROLLED, False, True],
+        [False, True, COURSE_VISIBILITY_PUBLIC, CourseUserType.ENROLLED, False, True],
+
+        [False, False, COURSE_VISIBILITY_PRIVATE, CourseUserType.UNENROLLED_STAFF, True, True],
+        [False, True, COURSE_VISIBILITY_PRIVATE, CourseUserType.UNENROLLED_STAFF, True, True],
+        [False, True, COURSE_VISIBILITY_PUBLIC_OUTLINE, CourseUserType.UNENROLLED_STAFF, True, True],
+        [False, True, COURSE_VISIBILITY_PUBLIC, CourseUserType.UNENROLLED_STAFF, True, True],
+
+        [False, False, COURSE_VISIBILITY_PRIVATE, CourseUserType.GLOBAL_STAFF, True, True],
+        [False, True, COURSE_VISIBILITY_PRIVATE, CourseUserType.GLOBAL_STAFF, True, True],
+        [False, True, COURSE_VISIBILITY_PUBLIC_OUTLINE, CourseUserType.GLOBAL_STAFF, True, True],
+        [False, True, COURSE_VISIBILITY_PUBLIC, CourseUserType.GLOBAL_STAFF, True, True],
     )
     @ddt.unpack
     def test_home_page(
-            self, enable_unenrolled_access, course_visibility, user_type,
+            self, invite_only, enable_unenrolled_access, course_visibility, user_type,
```

Comment:
I'm not sure adding a new test data makes sense, since it's only used for the `UNENROLLED` usertype. You can instead just set the flag within the test with something like `self.course.invitation_only = true` and verify that the correct message is displayed. Would be good to also include a check for the message displayed when the user is unenrolled and course is set to `invitation_only`.

Reply:
Could you explain abit? In order to make a course invitation only, I had to create private course in the setup().
I'm not sure how to set the `self.course.invitation_only = true` flag for the specific test cases here.
The actual test (L341-348) seems to test everything(course messages and user warning messages).

Reply:
What is being checked here is not really helpful. When you create the private course, you are setting the `invitation_only` flag to `True` and in the test case, you're passing the flag as the input and verifying that when the flag is True, the url points to the private course. 
For the scope of this task, you can just verify that the correct message text is displayed when the course has `invitation_only` set to true and the user is unenrolled.